prime-router/docs/proposals/0008-reporting-entity-front-end.md

```diff
@@ -0,0 +1,104 @@
+leveraging the existing tech stack including USDWS, Javascript, Eleventy, etc. We propose that based on a user’s
+permissions in Okta, it will determine which screens in the front end the user can see and access. In the case of the
+proposed CSV upload feature, a user with a permission of “sender” would see the CSV upload page on the ReportStream
+front end. The initial use-case for the CSV upload feature will be to meet an immediate need for a constrained list of
+senders who are uploading data into an unsecured Louisiana Department of Health web portal. The first user will be
+Greenlight Urgent Care who is able to systematically generate a CSV file from their Experity Electronic Medical Record
+System but has no way to get that data to ReportStream today.
+
+## Background
+
+As technical assistance teams are scaling up and out across the nation, the ReportStream’s web portal would be a good
+place for the assistance teams to send smaller, less tech-savvy submitting entities that need to get their data to the
+Public Health Department in a quick-and-(semi)-painless manner. One way to achieve this is to utilize ReportStream’s
+built-in features such as:
+
+  1. Authentication
+  2. ReportStream API integration 
+
+to create a form that accepts comma separated value (.CSV) files that can be
+sent to the ReportSteam API for the submitting entity’s respective Public Health Department to consume. This paper
+outlines a proposal on how this CSV Upload Submission feature would work and would be implemented.
+
+## Goals
+
+Create a new page on the ReportStream front-end which houses a simple web portal form that accepts CSV files, formatted
+in a constrained set of data formats for which ReportStream has sender schemas configured.
+
+This will involve segmenting users in the authentication service to allow for certain actions to be performed when given
+permissions to. For instance, if a submitting entity is also a State Public Health Department, they will have
+permissions to both see Daily Data (which is currently what they can see now), plus, they can have the permission to
+also Upload CSVs in their desired format. However, a submitting entity such as a hospital or school does not need the
+Daily Data screen, and so their permissions will solely be the Upload CSV feature.
+
+## Proposal
+
+We propose that to create a form in the ReportStream web portal that accepts CSV files for submitting entities to pass
+their data to their respective State Health Department.
+
+## Initial Implementation Plan
+
+This will involve 3 steps:
+1. **Authentication/Authorization:** The authentication service will have to be set up to send roles or permissions that
+pertain to the user/organization, and the web portal client, as well as the API, will need to consume these permissions
+and behave accordingly.
+
+    1. The permission that comes through can be called something like `sender`.
+
```

Comment:
Today, users receive their permissions through their membership in groups. I'd recommend that you follow this structure.